### PR TITLE
fixed formatting with clippy and removed Document trait

### LINKED ISSUE #255: Remove Document trait asking for UIDType
Currently to communicate a primary key on document addition the document structure must implement the following trait: 

```rust
// src/document.rs
pub trait Document: DeserializeOwned + std::fmt::Debug + Serialize {
    /// The type of the primary key
    type UIDType: Display;

    /// The method returning the primary key of the Document.
    ///
    /// **WARNING**! This method **MUST** only return an object that displays himself only using alphanumeric characters, '/' and '-'.
    /// Otherwise, the Meilisearch server will reject your document.
    fn get_uid(&self) -> &Self::UIDType;
}
```

Here `UIDType` is the primary key.  This trait is not very intuitive and makes the code harder to use than it should be. 
The naming is not very clear and it requires an additional step of implementation on all document addition even when the user does not need to communicate the primary key. 

for example: 
```rust
  // Define the type of our documents
  #[derive(Serialize, Deserialize, Debug)]
  struct Movie {
    id: String,
    title: String
  }
  impl Document for Movie {
    type UIDType = String;
    fn get_uid(&self) -> &Self::UIDType { &self.id }
  }

  // Add a document to our index
  let task: Task = client.index("movies").add_documents(&[
    Movie {
      id: "123sq178".to_string(),
      title: "Amélie Poulain".to_string(),
    }
  ], None).await.unwrap();
```
and when the user wants to communicate the primary key he still needs to give it as the second parameter of add_documents

```rust
add_documents(documents, Some("id"))
```

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request
Since @honne23 doesn't seem available currently, and we have #262, #263, and #264 waiting for this PR to be merged, I took back on it.

But thanks a lot, @honne23; you wrote most of the code, and I just applied the few left comments.
Thanks for contributing!

## What does this PR do?
Fixes #255 
Closes #261

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by sanders41** [comment]
@irevoire just an FYI based on your comment, you can add people as a co-author of the PR if you want.

https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line

**Comment by irevoire** [comment]
Oh, thank you! I did that :grin:

File: src/indexes.rs [hunk 1]
```diff
@@ -464,7 +422,7 @@ impl Index {
     /// # movie_index.delete().await.unwrap().wait_for_completion(&client, None, None).await.unwrap();
     /// # });
     /// ```
-    pub async fn add_or_update<T: Document>(
+    pub async fn add_or_update<T: DeserializeOwned + Serialize + std::fmt::Debug>(
```
[the remark this answers is not in the document]
Comment (irevoire) [reply src/indexes.rs]: Ah yes good catch I missed that one!

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -807,15 +778,11 @@ security_guide_delete_key_1: |-
 landing_getting_started_1: |-
   let client = Client::new("http://localhost:7700", "masterKey");
 
-  #[derive(Serialize, Deserialize, Debug)]
+  #[derive(Serialize, Deserialize )]
```
Comment (brunoocasali) [inline .code-samples.meilisearch.yaml:781]: Extra space detected 🤖

File: src/settings.rs [hunk 3]
```diff
@@ -209,19 +209,19 @@ impl Index {
     /// # });
     /// ```
     pub async fn get_settings(&self) -> Result<Settings, Error> {
-        Ok(request::<(), Settings>(
+        request::<(), Settings>(
```
Comment (brunoocasali) [inline src/settings.rs:212]: Can you give me some idea behind this change? Why it was required before and now isn't?

**Review by brunoocasali — APPROVED** [review]
Thanks for your contribution @honne23 and @irevoire you're the best! :)

**Comment by brunoocasali** [comment]
bors merge

**Comment by brunoocasali** [comment]
bors merge

**Pull request merged**